### ISSUE TITLE
Use Billing Dates for Hospitalization operator

### ISSUE THREAD

**Issue by jenniferduryea** [outcomesinsights/conceptql#247]
After #194 and #197, it looks like users may want to query and report the events in a different way.  Please see specs below.  The changed areas are under output for gdm.

### Logic for GDM

- Look for `collections` records
  - `collections` record must have
    - associated `admission_details` record
    - `collections.collection_type_concept_id` is mapped to "inpatient" or "snf"

### Logic for OMOPv4.5+

- `visit_occurrence` records
  - `visit_occurrence.place_of_service_concept_id` = any "inpatient" type concept

## Output

Desired output columns:

- Length of stay
- Admission source
- Discharge status
- Primary diagnosis code
- Primary diagnosis vocabulary
- Admit date
- Discharge date
- billing start date
- billing end date

### Columns used for output in GDM

- Need the following records to assemble those columns:
  - `admission_details` as `collections.admission_details`
  - `primary_clinical_code` as `collections.clinical_codes.provenance_concept_id` = "primary"
- Length of stay
  - (`admission_details.discharge_date` - `admission_details.admission_date`) + 1
- Admission source
  - `admission_details.admit_source_concept_id.concept_text`
- Discharge status
  - `admission_details.discharge_location_concept_id.concept_text`
- Primary diagnosis code
  - `primary_clinical_code.clincal_code_concept.concept_code`
- Primary diagnosis vocabulary
  - `primary_clinical_code.clincal_code_concept.vocabulary_id`
- Admit date
  - `admission_details.admission_date`
- Discharge date
  - `admission_details.discharge_date`
- Billing Start date
  - `collections.start_date`
- Billing End date
  - `collections.end_date`

### Columns used for output in OMOPv4.5+

- Length of stay
  - (`visit_occurrence.visit_end_date` - `visit_occurrence.visit_start_date`)
- Admission source
  - NULL
- Discharge status
  - NULL
- Primary diagnosis code
  - NULL
- Primary diagnosis vocabulary
  - NULL
- Admit date
  - `visit_occurrence.visit_start_date`
- Discharge date
  - `visit_occurrence.visit_end_date`
- Billing Start date
  - NULL
- Billing End date
  - NULL

For output, the Billing Start and Billing End date will be specified in Jigsaw output as events.start_date and events.end_date.  Primary Code and Primary Diagnosis Vocabulary will use events.source_value and events.source_vocabulary_id.  All of the other fields will be new columns.

Since we have two sets of dates (billing and admit/discharge), conceptql will use the Billing Dates for the start/end date operations.  And the Protocol Builder will use billing dates for event selection.

Tagging @aguynamedryan for implementation and @markdanese for any discussion.

**Comment by aguynamedryan** [comment]
@jenniferduryea, a couple of questions:

- If billing start/end_date are going to be used as the start/end_date of ConceptQL's results, under OMOPv4.5+, these results will _always_ be NULL and never work with temporal operators.   Is that the desired behavior?
- Your comments make reference to Primary Code/Diagnosis columns.  It doesn't look like their behavior changes in this ticket from #194.  Why are they discussed?

**Comment by jenniferduryea** [comment]
@aguynamedryan 
- No.  Billing start/end date will refer to visit_occurrence_start_date and visit_occurrence_end_date.  OMOP will use just visit_occurrence dates for both billing dates and admit dates.
- just to be complete. I like the whole schematic referred to in one ticket.

**Comment by aguynamedryan** [comment]
@jenniferduryea, is it ok if these changes also affect SNF (#198)?

**Comment by jenniferduryea** [comment]
@aguynamedryan oh yeah!  Good catch. Yes.

**Comment by jenniferduryea** [comment]
Assigning @marc-outins to this for implementation.

**Comment by marc-outins** [comment]
@jenniferduryea for hospice, snf, and hospitalization operators they should use start/end dates = billing dates and then add admission/discharge dates as separate columns in the events file correct?

**Comment by jenniferduryea** [comment]
@marc-outins correct

**Comment by marc-outins** [comment]
@jenniferduryea this is ready to test on chisel.jsaw.io

**Comment by jenniferduryea** [comment]
This was a bit difficult to test because it's rare to have hospitalization billing dates that are not equal to admit/discharge dates.  I've been informed that this code used to have this is also relevant to hospice and snf operators.  So i tested this on hospice data using medicare lds dataset on titan.  I confirmed that the billing dates (i.e. start and end dates in collections) are being used by tshank for evaluation. I also ensured that admit and discharge dates for the associated admission_detail_ids are also filled out as separate columns under the Jigsaw events output table.  Looks great!  Closing.
